core/camel-support/src/main/java/org/apache/camel/support/CamelContextHelper.java

```diff
@@ -587,4 +590,53 @@ public static String getRouteId(NamedNode node) {
         return parent != null ? parent.getId() : null;
     }
 
+    /**
+     * Gets the {@link RestConfiguration} from the {@link CamelContext} and check if the component which consumes the
+     * configuration is compatible with the one for which the rest configuration is set-up.
+     *
+     * @param camelContext the camel context
+     * @param component the component that will consume the {@link RestConfiguration}
+     * @return the {@link RestConfiguration}
+     * @throws IllegalArgumentException is the component is not compatible with the {@link RestConfiguration} set-up
+     */
+    public static RestConfiguration getRestConfiguration(CamelContext camelContext, String component) {
+        RestConfiguration configuration = camelContext.getRestConfiguration();
+
+        validateRestConfigurationComponent(component, () -> configuration.getComponent());
+
+        return configuration;
+    }
+
+    /**
+     * Gets the {@link RestConfiguration} from the {@link CamelContext} and check if the component which consumes the
+     * configuration is compatible with the one for which the rest configuration is set-up.
+     *
+     * @param camelContext the camel context
+     * @param component the component that will consume the {@link RestConfiguration}
+     * @param producerComponent the producer component that will consume the {@link RestConfiguration}
+     * @return the {@link RestConfiguration}
+     * @throws IllegalArgumentException is the component is not compatible with the {@link RestConfiguration} set-up
+     */
+    public static RestConfiguration getRestConfiguration(CamelContext camelContext, String component, String producerComponent) {
+        RestConfiguration configuration = camelContext.getRestConfiguration();
+
+        validateRestConfigurationComponent(component, () -> configuration.getComponent());
+        validateRestConfigurationComponent(producerComponent, () -> configuration.getProducerComponent());
+
+        return configuration;
+    }
+
+    private static void validateRestConfigurationComponent(String component, Supplier<String> configurationComponentFn) {
```

Comment:
Instead of the Suppler then just use a String parameter. It doesnt make sense to use a function here IMHO.

Reply:
I does not, I've re-factored that bit many times so the final iteration is simpler and does not require any fancy stuff.

Thx for spotting it